databuilder/models/table_metadata.py

```diff
@@ -457,6 +470,13 @@ def _create_next_relation(self) -> Iterator[GraphRelationship]:
                 )
                 yield tag_relationship
 
+        if self.badges:
+            table_badge_metadata = BadgeMetadata(start_label=TableMetadata.TABLE_NODE_LABEL,
+                                                 start_key=self._get_table_key(),
+                                                 badges=self.badges)
+            for relation in table_badge_metadata.create_relation():
```

Comment:
Oh yeah - good catch, this?: https://www.amundsen.io/amundsen/tutorials/badges/#badges-configuration - I see that what I added will just fall back to `BadgeStyle.DEFAULT` on the frontend... I must admit at first my focus was just two things:

1. To restore the old (table level) badge behavior from when it was a kid of tags - also in the Quickstart.
2. Get the delicious column level badges surfaced! 😍 

So I didn't give this a lot of thought till now.

I guess it's not entirely clear to me what having the categories in the metadata as well as defining it (more staticly) on the frontend will drive for the end user? To me it seems I could get away with just configuring the mapping on the frontend for e.g. "alpha" and "beta" badges - and the visual categorization would be implied.

----

Also _now_ (some hours later) mulling _more_ over this and after re-reading https://github.com/amundsen-io/rfcs/blob/master/rfcs/001-support-badge-as-entity.md it's still not clear to me what `category` drives. Will it eventually light something up in search - or something else which I didn't notice and can't read out of the RFC? And it even seems that I got away with finding a loophole as Databuilder currently does not enforce `category` to be defined (unsure if it **should** do that?). But I'm confident that the specific use cases you have at Lyft, and perhaps of you already internally had some future plans could clarify. More better/examples usually casts a lot of light on things 🙂 Crux is of course someone has to spend time on providing them.

----

The tl;dr

Two main points:
* So all in all I think I just surfaced a few things that could be clarified - and also explained where I came from with what I implemented. I don't think none of that is blocking for merging neither #417 nor #418, as soon as you think they're good enough to be merged.
* I think both #417 and #418 complements each other and each could have their use cases. #417 for the simple stuff #418 for the full thing.

What do you think? Next step?

Reply:
I was thinking we could keep the behavior here that adds column level badges since all column badges fall into the same category of "column". And we can use #417 for table level badges. Categories basically serve are an additional field that can serve to indicate something like the placement of a badge ('column', 'table') or any other future use. What do you think?

Reply:
OK, I’ll back out the table level stuff then when I get near my computer